### ISSUE TITLE
Running out of CircleCI credits

### ISSUE THREAD

**Issue by rth** [pyodide/pyodide#4224]
I guess you also received this message,

> Your team on pyodide ran out of 30k monthly credits offered on our Free plan and can no longer run tests, builds, and deploys for the rest of this month.
>
> In the last 30 days your team used the Free plan to test, build, and deploy for approximately 102518 minutes a day. That's a lot of building.
>
> Without additional credits, your team will not be able to continue until the next refill on 11/08/23.
>
> Want to immediately access more credits and unblock your team?
>
> Click below to upgrade to the Performance plan, starting at just $15/month. 

The [current usage](https://app.circleci.com/settings/plan/github/pyodide/overview) is a bit confusing (and there is some of contradictory information: "102518 minutes a day" ?? ). Our limit is 400,000 credits / month not 30k I think.

![image](https://github.com/pyodide/pyodide/assets/630936/227bca48-a832-4ff1-90e8-e19d1d0e629d)

But yeah, a few 2-4k$ per month is what I would say our current CI usage is, if we had to pay for it.
I guess short term we could survive with just GH Actions building the core suite, but we cannot make a release that way.

Overall though I think we should stop building all packages on every commit. We could also try asking them to increase the limit, but I feel like our current way is rather wasteful and a bit hard to justify.

**Comment by ryanking13** [comment]
Yes, I also got the mail.

I think we should start working on separating package builds from the main repository, as discussed in https://github.com/pyodide/pyodide/pull/3827#issuecomment-1539824379. I hadn't touched it yet because I thought it would be a pretty big task, but now it looks like we need to start working on it...

**Comment by rth** [comment]
Agreed. However, it looks like CircleCI  builds are still running on main and it's not a hard limit (at least for us), though we should probably still address this.

**Comment by ryanking13** [comment]
Okay. I started working on the https://github.com/ryanking13/pyodide-recipes-mirror repository again. For now, the goal is to move all the recipes to that repository and build them using xbuildenv, and I'm currently successful in building numpy and scipy (which is very promising). I'll transfer it to pyodide organization when I think it is ready (maybe with name `pyodide-recipes` or `pyodide-packages`)?

**Comment by rth** [comment]
Thanks @ryanking13 !

**Comment by ryanking13** [comment]
We haven't seen this error in a while, and since we've unvendored recipes, the CircleCI credit usage should decrease significantly. Closing as resolved.
